elephant/change_point_detection.py

```diff
@@ -0,0 +1,514 @@
+        alpha : integer
+            alpha-quantile for the set of maxima of the limit processes
+        n_surrogates : integer
+            numbers of simulated limit processes
+        dt : quantity
+          resolution, time step at which the windows are slided
+        test_param : np.array 3 * num of window,
+            first row: list of `h`, second and third rows: empirical means and
+            variances of the limit process correspodning to `h`. This will be 
+            used to normalize the `filter_process` in order to give to the every
+            maximum the same impact on the global statistic.
+           
+
+    Returns:
+    --------
+        cps : list of list
+           one list for each `h`, containing the points detected with the
+           corresponding `filter_process`. N.B.: only cps whose h-neighborhood 
+           does not include previously detected cps (with smaller window h) are
+           added to the list.
+    """
+
+    if (test_quantile is None) & (test_param is None):
+        test_quantile, test_param = empirical_parameters(window_sizes, t_final,
+                                                         alpha, n_surrogates,
+                                                         dt)
+    elif test_quantile is None:
+        test_quantile = empirical_parameters(window_sizes, t_final, alpha,
+                                             n_surrogates, dt)[0]
+    elif test_param is None:
+        test_param = empirical_parameters(window_sizes, t_final, alpha,
+                                          n_surrogates, dt)[1]
+                                          
+    spk = spiketrain
+    
+    #  List of list of detected point, to be returned
+    cps = []  
+    
+    for i, h in enumerate(window_sizes):
+        dt_temp = dt
+        # automatic setting of dt
+        if dt_temp is None:  
+            dt_temp = h / 20.
+        # filter_process for window of size h
+        t, differences = _filter_process(dt_temp, h, spk, t_final, test_param)
+        time_index = np.arange(len(differences))
+        # Point detected with window h
+        cps_window = []
+        while np.max(differences) > test_quantile:
+            cp_index = np.argmax(differences)
+            # from index to time
+            cp = cp_index * dt_temp + h  
+            #print("detected point {0}".format(cp), "with filter {0}".format(h))
+            # before to repet the procedure the h-neighbourg of 'cp' detected 
+            # is cut, because rate changes within it are explained by this cp
+            differences[np.where(
+                (time_index > cp_index - int(h / dt_temp.rescale(h.units))) & (
+                        time_index < cp_index + int(
+                            h / dt_temp.rescale(h.units))))[0]] = 0
+            # check if the neighbourhood of detected cp does not contain cps 
+            # detected with other windows
+            neighbourhood_free = True
+            if i == 0:
```

Comment:
This if-case is not necessary.
When `i == 0`, the for-loop in the `else` case actually doesn't iterate anything.